packages/react-focus/src/components/FocusZone/FocusZone.tsx

```diff
@@ -178,7 +178,9 @@ export class FocusZone extends React.Component<IFocusZoneProps> implements IFocu
     if (
       doc &&
       this._lastIndexPath &&
-      (doc.activeElement === doc.body || doc.activeElement === root || doc.activeElement === null)
+      (doc.activeElement === doc.body ||
+        doc.activeElement === null ||
+        (this.props.restoreFocusFromRoot && doc.activeElement === root))
```

Comment:
looks good to me, but pls note that this changes the behavior for v7 slightly as the prop would be false by default

Reply:
@jurokapsiar I was discussing with @khmakoto , really glad you called this out.

Today if focus is on a button in the zone and a render causes the button to go away, focus will be moved to the next button. With this change it would just go to the body.

We talked about whether this should be opt-in or opt-out behavior. I think in most cases you'd want focus to be restored. There are some cases where the consume is managing focus and doesn't want the behavior, but i think that would be minority scenarios.

I'd propose we change this to `preventFocusRestoration` to opt out if it causes problems. This would enable us to ship this change now without a fundamental break, but would mean that for scenarios where that causes issues, there's a way to opt out.

What do you think?

Reply:
sorry for late response. I agree. If I remember correctly, this was causing some issues when focus zone was in a focus trap zone, but opting out gives us a way to handle this case.